react/utils/client/index.ts

```diff
@@ -78,9 +82,11 @@ export const getClient = (runtime: RenderRuntime, baseURI: string, runtimeContex
 
     const uriSwitchLink = createUriSwitchLink(baseURI, workspace)
 
+    const baseLink = [omitTypenameLink, versionSplitterLink, runtimeContextLink, ensureSessionLink, persistedQueryLink, uriSwitchLink]
+
     const link = cacheControl
-      ? ApolloLink.from([omitTypenameLink, versionSplitterLink, runtimeContextLink, ensureSessionLink, persistedQueryLink, uriSwitchLink, cachingLink(cacheControl), fetcherLink])
-      : ApolloLink.from([omitTypenameLink, versionSplitterLink, runtimeContextLink, ensureSessionLink, persistedQueryLink, uriSwitchLink, fetcherLink])
+      ? ApolloLink.from([...baseLink, cachingLink(cacheControl), fetcherLink])
```

Comment:
Maybe create a separate array for `cachingLink` which optionally has this link or is empty, and always spread it here. (So we avoid two similar arrays)